src/__tests__/waitFor.test.js

```diff
@@ -87,9 +91,24 @@ test('works with fake timers', async () => {
   } catch (e) {
     // suppress
   }
-  jest.runTimersToTime(400);
+  jest.advanceTimersByTime(400);
 
   expect(mockFn).toHaveBeenCalledTimes(3);
+});
 
-  jest.useRealTimers();
+test('works with modern fake timers', async () => {
```

Comment:
Similar as in autocleanup: test functions for default, modern and legacy timers.